src/Monad/Reader.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Widmogrod\Monad;
+
+use Widmogrod\Common;
+use Widmogrod\FantasyLand;
+
+class Reader implements FantasyLand\Monad
+{
+    const of = 'Widmogrod\Monad\Reader::of';
+
+    use Common\PointedTrait;
+
+    public function bind(callable $function)
+    {
+        return self::of(function ($env) use ($function) {
+            return $function($this->runReader($env))->runReader($env);
```

Comment:
on some closures format before PHP7 this direct invocation will fail.
Please use 
```php
call_user_func($function, $this->runReader($env))
```